src/test/java/tests/HandlerTest.java

```diff
@@ -1,70 +1,145 @@
 package tests;
 
 import static org.junit.Assert.fail;
-import httpserver.HTTPException;
-import httpserver.HTTPHandler;
-import httpserver.HTTPRequest;
-import httpserver.HTTPResponse;
-import httpserver.HTTPRouter;
+import static org.junit.Assert.assertEquals;
+import httpserver.HttpException;
+import httpserver.HttpHandler;
+import httpserver.HttpRequest;
+import httpserver.HttpResponse;
+import httpserver.HttpRouter;
+import httpserver.HttpServer;
+import httpserver.Route;
+
+import java.util.Map;
+import java.net.ServerSocket;
 
 import org.junit.Test;
+import org.junit.BeforeClass;
+
+import tests.mocks.MockHttpServer;
+import tests.mocks.MockClient;
+
+public class HandlerTest {
+    private static HttpServer server;
+
+    @BeforeClass
+    public static void setupServer() {
```

Comment:
Maybe we should think of some better way to abstract this out?

This is actually probably fine, I'm just thinking too far out there. We can figure out an abstraction layer later if we actually need it, and just write a little boilerplate for each test class.